packages/taquito/src/prepare/prepare-provider.ts

```diff
@@ -1098,4 +1101,78 @@ export class PrepareProvider extends Provider implements PreparationProvider {
       contents,
     };
   }
+
+  /**
+   *
+   * @description Method to prepare a stake operation
+   * @param operation RPCOperation object
+   * @param source string or undefined source pkh
+   * @returns a PreparedOperation object
+   */
+  stake(operation: StakingParams, source?: string): Promise<PreparedOperation> {
+    return this.staking(operation, 'stake', source);
+  }
+
+  /**
+   *
+   * @description Method to prepare an unstake operation
+   * @param operation RPCOperation object
+   * @param source string or undefined source pkh
+   * @returns a PreparedOperation object
+   */
+  async unstake(operation: StakingParams, source?: string): Promise<PreparedOperation> {
+    return this.staking(operation, 'unstake', source);
+  }
+
+  /**
+   *
+   * @description Method to prepare a finalize_unstake operation
+   * @param operation RPCOperation object
+   * @param source string or undefined source pkh
+   * @returns a PreparedOperation object
+   */
+  async finalizeUnstake(operation: StakingParams, source?: string): Promise<PreparedOperation> {
+    return this.staking(operation, 'finalize_unstake', source);
+  }
+
+  private async staking(
+    { fee, storageLimit, gasLimit, ...rest }: StakingParams,
+    entrypoint: StakingEntrypoint,
+    source?: string
```

Comment:
Like previous comments should `source?` be in `StakingParams` and `entrypoint` be under `parameter: TransactionOperationParameter`